cardano-cli/src/Cardano/CLI/Options/Debug/Transaction.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+{- HLINT ignore "Use <$>" -}
+{- HLINT ignore "Move brackets to avoid $" -}
+
+module Cardano.CLI.Options.Debug.Transaction
+  ( parseDebugCmds
+  ) where
+
+import           Cardano.Api.Shelley hiding (QueryInShelleyBasedEra (..))
+
+import           Cardano.CLI.Commands.Debug
+import           Cardano.CLI.Commands.Debug.LogEpochState
+import           Cardano.CLI.Environment
+import           Cardano.CLI.EraBased.Options.Common
+import           Cardano.CLI.Options.Debug.Transaction.Echo
+
+import           Data.Foldable
+import           Options.Applicative hiding (help, str)
+import qualified Options.Applicative as Opt
+
+parseDebugCmds :: EnvCli -> Parser DebugCmds
+parseDebugCmds envCli =
+  Opt.hsubparser $ mconcat
+    [ Opt.metavar "debug transaction commands"
+    , Opt.commandGroup "debug commands"
+    , Opt.command "transaction"
+        $ Opt.info (pDebugCmds envCli)
+        $ Opt.progDesc "Debug transaction commands"
+    ]
+
+pDebugCmds :: EnvCli -> Parser DebugCmds
+pDebugCmds envCli =
+   asum
+     [ subParser "log-epoch-state"
+         $ Opt.info pLogEpochStateCmdArgs
+         $ Opt.progDesc
+         $ mconcat
+            [ "Log epoch state of a running node."
+            , " This command will connect to a local node and log the epoch state to a file."
+            , " The log file format is line delimited JSON."
+            , " The command will not terminate."
+            ]
+    , subParser "echo"
+        $ Opt.info (DebugTransactionCmds <$> pDebugTransactionEcho)
+        $ Opt.progDesc "Echo a transaction"
```

Comment:
Can you add more info why specifically we are adding this? I can approve after.

Reply:
I was going to say the same. Something like "it parses and rewrites the transaction" or "it deserialises and then reserialises a transaction" or "it re-formats the transaction". And it would be nice to say why this may be useful too, e.g: "because we may want to update/reformat the transaction using the format of the current version of the cli".